plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/util/SynthesizedColumnHandler.java

```diff
@@ -238,4 +251,68 @@ public long getFileModificationTime()
             return modifiedTime;
         }
     }
+
+    /**
+     * Helper function to prefill BlockBuilders with values from PartitionKeys which are in the string type.
+     * This function handles the casting of String type the actual column type.
+     */
+    private static void appendToBuilder(Type type, Object value, BlockBuilder blockBuilder)
+    {
+        if (value == null) {
+            blockBuilder.appendNull();
+            return;
+        }
+
+        if (type instanceof VarcharType varcharType) {
+            varcharType.writeSlice(blockBuilder, utf8Slice(value.toString()));
+        }
+        else if (type instanceof IntegerType integerType) {
+            integerType.writeInt(blockBuilder, Integer.parseInt((String) value));
+        }
+        else if (type instanceof BigintType bigintType) {
+            bigintType.writeLong(blockBuilder, Long.parseLong((String) value));
+        }
+        else if (type instanceof BooleanType booleanType) {
+            booleanType.writeBoolean(blockBuilder, Boolean.parseBoolean((String) value));
+        }
+        else if (type instanceof DecimalType decimalType) {
+            SqlDecimal sqlDecimalFromStr = SqlDecimal.decimal((String) value, decimalType);
+
+            if (decimalType.isShort()) {
+                // For short decimals, get the unscaled long value
+                // SqlDecimal.toBigDecimal() is used for consistency with the original SqlDecimal path
+                // The unscaled value of a Trino short decimal (precision <= 18) fits in a long
+                writeShortDecimal(blockBuilder, sqlDecimalFromStr.toBigDecimal().unscaledValue().longValue());
+            }
+            else {
+                // For long decimals, use the BigDecimal representation obtained from SqlDecimal.
+                writeBigDecimal(decimalType, blockBuilder, sqlDecimalFromStr.toBigDecimal());
```

Comment:
nit: `sqlDecimalFromStr.toBigDecimal()` is shared so define it to replace `sqlDecimalFromStr` for reuse?

Reply:
Done